src/Core/Tools/Models/Business/ReferenceEvent.cs

```diff
@@ -25,7 +25,7 @@ public class ReferenceEvent
     /// <param name="type">Monitored event type.</param>
     /// <param name="source">Entity that created the event.</param>
     /// <param name="currentContext">The conditions in which the event occurred.</param>
-    public ReferenceEvent(ReferenceEventType type, IReferenceable source, ICurrentContext currentContext)
+    public ReferenceEvent(ReferenceEventType type, IReferenceable? source, ICurrentContext? currentContext)
```

Comment:
❌ `source` and `currentContext` check nullability of their arguments for historical purposes. They should never be null. Please adjust the calling code accordingly.

```suggestion
    public ReferenceEvent(ReferenceEventType type, IReferenceable source, ICurrentContext currentContext)
```

Reply:
@audrey-jensen , ah, oops. I saw the null checks and thought it would be safe to make them nullable. Sorry about that!

For my new `ReferenceEvent`, I do not have any entity which implements `IReferenceable` with a guid during the start registration process. Is there an alternative or something I'm missing that can help me achieve firing a reference event at that point in time?

Reply:
Oh, interesting. This doesn't actually throw like I thought it would:
![image](https://github.com/bitwarden/server/assets/116684653/9123ee4e-2113-4731-bdee-81f7bfbdc758)

Ok, I've reverted my change!

Reply:
Hmm... this is a rather unique situation. I'd probably use the default constructor. You'll need to [create a new `ReferenceEventSource` member](https://github.com/bitwarden/server/blob/main/src/Core/Tools/Enums/ReferenceEventSource.cs#L5). 

Once that's done, you'll need to set the `Type` and `Source` fields manually. The `Id` field will initialize to `Guid.Empty`.

Reply:
Ok, thank you! I've refactored to use the default constructor!